docs/compilers/Deterministic Inputs.md

```diff
@@ -0,0 +1,26 @@
+Deterministic Inputs
+====================
+
+We are aiming to make the compilers ultimately deterministic (https://github.com/dotnet/roslyn/issues/372). What that means is that the "same inputs" will cause the compilers to produce the "same outputs". 
+
+The following will be considered inputs to the compiler for the purpose of determinism:
+
+- The sequence of command-line flags
+- The contents of the compiler's `.rsp` response file.
+- The precise version of the compiler used, and its referenced assemblies
+- Current full directory path (you can reduce this to a relative path; see https://github.com/dotnet/roslyn/issues/949)
+- (Binary) contents of all files explicitly passed to the compiler, directly or indirectly, including
+  - source files
+  - referenced assemblies
+  - referenced modules
+  - resources
+  - the strong name key file
+  - `@` response files
+  - Analyzers
```

Comment:
I think of that as an example of "Analyzers" passed to the compiler "indirectly". But it is worth calling out.